lighthouse-core/report/v2/report-renderer.js

```diff
@@ -19,19 +19,190 @@
+      case 'text': return this._renderText(details);
+      case 'block': return this._renderBlock(details);
+      case 'list': return this._renderList(details);
+      default: throw new Error('Unknown type: ' + details.type);
+    }
+  }
+
+  /**
+   * @param {DetailsJSON} text
+   * @return {Element}
+   */
+  _renderText(text) {
+    const element = this._createElement('div', 'lighthouse-text');
+    element.textContent = text.text;
+    return element;
+  }
+
+  /**
+   * @param {DetailsJSON} block
+   * @return {Element}
+   */
+  _renderBlock(block) {
+    const element = this._createElement('div', 'lighthouse-block');
+    for (const item of block.items)
+      element.appendChild(this._renderDetails(item));
+    return element;
+  }
+
+  /**
+   * @param {DetailsJSON} list
+   * @return {Element}
+   */
+  _renderList(list) {
+    const element = this._createElement('div', 'lighthouse-list');
+    if (list.header) {
+      const header = this._createChild(element, 'div', 'lighthouse-list-header');
+      header.appendChild(this._renderDetails(list.header));
+      header.addEventListener('click', () => items.classList.toggle('hidden'), false);
+    }
+    const items = this._createChild(element, 'div', 'lighthouse-list-items hidden');
+    for (const item of list.items)
+      items.appendChild(this._renderDetails(item));
+    return element;
+  }
+
+  /**
+   * @param {Error} e
+   * @return {Element}
+   */
+  _renderException(e) {
+    const element = this._createElement('div', 'lighthouse-exception');
+    element.textContent = String(e);
+    return element;
+  }
+
+  /**
+   * @param {ReportJSON} report
+   * @return {Element}
+   */
+  _renderReport(report) {
+    const element = this._createElement('div', 'lighthouse-report');
+    for (const category of report.reportCategories)
+      element.appendChild(this._renderCategory(category));
+    return element;
+  }
+
+  /**
+   * @param {CategoryJSON} category
+   * @return {Element}
+   */
+  _renderCategory(category) {
+    const element = this._createElement('div', 'lighthouse-category');
+    element.appendChild(this._renderScore(category.score, category.name, category.description));
+    for (const audit of category.audits)
+      element.appendChild(this._renderAudit(audit));
+    return element;
+  }
+
+  /**
+   * @param {AuditJSON} audit
+   * @return {Element}
+   */
+  _renderAudit(audit) {
+    const element = this._createElement('div', 'lighthouse-audit');
+    let title = audit.result.description;
+    if (audit.result.displayValue)
+      title += ': ' + audit.result.displayValue;
+    element.appendChild(this._renderScore(audit.score, title, audit.result.helpText));
+    if (audit.result.details)
+      element.appendChild(this._renderDetails(audit.result.details));
+    return element;
   }
 };
 
+// /** @typedef {{type: string, text: string|undefined, header: DetailsJSON|undefined, items: Array<DetailsJSON>|undefined}} */
+// var DetailsJSON;
```

Comment:
> My vote is for 4

SGTM2.

We were also just talking and realized it doesn't matter if this is `var` or `let` here anyways, since even if we use `var` here if somewhere else a global with the same name is declared with `let` or `const` it will throw an error. So I'm good with `let` and global pollution.